idaes/generic_models/unit_models/distillation/tray.py

```diff
@@ -547,7 +547,46 @@ def rule_flow(self, t, i):
                                           rule=rule_flow)
                         self.add_component("e_flow_" + port.local_name,
                                            expr)
-                        port.add(expr, k)
+                        port.add(expr, local_name)
+                elif "phase" in local_name:
+                    # flow is indexed by phase and comp
+                    # Get the indexing sets i.e. component list and phase list
+                    component_set = self.config.\
+                        property_package.component_list
+
+                    phase_set = self.config.\
+                        property_package.phase_list
+
+                    def rule_flow(self, t, p, i):
+                        if (phase is self._liquid_set and
+                                p in self._liquid_set) or \
+                                (phase is self._vapor_set and
+                                 p in self._vapor_set) :
+                            # pass appropriate phase flow values to port
+                            return (self.properties_out[t].
+                                    component(local_name)[p, i]) * (side_sf)
+                        else:
+                            # return small number for phase that should not
+                            # be in the appropriate port. For example,
+                            # the state vars will be flow_mol_phase_comp
+                            # which will include all phases. The liq port
+                            # should have the correct references to the liq
+                            # phase flow but the vapor phase flow should be 0.
+                            return 1e-8
+
+                    expr = Expression(self.flowsheet().time,
+                                      phase_set,
+                                      component_set,
+                                      rule=rule_flow)
+                    self.add_component("e_" + local_name + port.local_name,
+                                       expr)
+                    port.add(expr, k)
+                else:
+                    raise PropertyNotSupportedError(
```

Comment:
I would suggest a PropertyPackageError rather than ProeprtyNotSupported - the "not supported" error is intended to indicate that a unit model asked for a property that the property package did not support. This looks like the opposite however - the property package told a unit model to use something the unit model does not know how to handle.